File: AndroidAacRxJava/app/src/main/java/com/example/raghu/androidaacrxjava/di/ActivityBuilder.java
package com.example.raghu.androidaacrxjava.di;



import com.example.raghu.androidaacrxjava.MainActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

/**
 * Created by raghu on 26/9/17.
 */

@Module
public abstract class ActivityBuilder {

    @ContributesAndroidInjector
    abstract MainActivity contributeMainActivity();
}

File: Dagger2Kotlin/app/src/main/java/com/example/raghu/dagger2testandroid/dagger/ActivityBuilder.kt
package com.example.raghu.dagger2testandroid.dagger

import com.example.raghu.dagger2testandroid.ui.main.MainActivity
import com.example.raghu.dagger2testandroid.ui.main.MainActivityModule

import dagger.Module
import dagger.android.ContributesAndroidInjector

/**
 * Created by raghu on 4/8/17.
 */

@Module
abstract class ActivityBuilder {

    @ContributesAndroidInjector(modules = arrayOf(MainActivityModule::class))
    internal abstract fun bindMainActivity(): MainActivity


}